test/ui/suite/command-palette.ts

```diff
@@ -0,0 +1,37 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+/*
+TODO:
+Parser for available commands in package.json
+Test individual commands
+*/
+
+import { expect } from 'chai';
+import { InputBox, Workbench } from 'vscode-extension-tester';
+import { COMMANDS } from '../common/constants';
+
+export function checkCommandPaletteOffering() {
+    describe('Command Palette', () => {
+        let prompt: InputBox;
+
+        before(async () => {
+            prompt = await new Workbench().openCommandPrompt() as InputBox;
+            await prompt.setText('>OpenShift');
+        });
+
+        it('Shows available commands', async () => {
+            expect((await prompt.getQuickPicks()).length > 1);
+        });
+
+        describe('Available Commands', () => {
+            COMMANDS.forEach(command =>
+                it(command, async () => {
+                    expect(await prompt.findQuickPick(command)).not.undefined;
```

Comment:
This looks like you are testing if your COMMANDS array is correct. It does not add any value to assure quality. If developers of extension decide to remove one of your COMMANDS form palette, this test fails and it does not mean you found a problem in extension.

What would make sense here is to verify, for example, if a command declared in package.json does not have a actual command handler registered when extension is activated. But that is already covered in UI tests and does not need verification using UI test.

Reply:
Exactly as you say. Further plan for this was to pick commands from the package.json instead of the COMMANDS array. The solution with the array is only temporary to have at least some test.

Reply:
Ok, but what are you going to test? If you going to read package.json and analyze if command should appear in palette it is a waste of processing time, because that is supposed to be tested by Microsoft and test is going to work always. If extension is activated all the commands guaranteed to be loaded and appear on pallet if they configured to appear.

It does not mean we do not need UI tests at all, I am not saying that. I am saying we covered tests related to commands verification and you should not waste your time on that.

Reply:
This PR can be closed then. Can you provide more info about this issue https://github.com/redhat-developer/vscode-openshift-tools/issues/2483 if you already covered tests related to commands verification?

Reply:
Done [here](https://github.com/redhat-developer/vscode-openshift-tools/issues/2483#issuecomment-1373999005)